### PR TITLE
UIDATIMP-1151: Fix unit tests for the ViewAllLogs component

### PR DESCRIPTION
**Pull request by jakhongirabdukhamdiov**
## Purpose
Fix unit tests for the ViewAllLogs component

## Refs
[UIDATIMP-1151](https://issues.folio.org/browse/UIDATIMP-1151)

### CODE REVIEW

**Review by OleksandrHladchenko1 — APPROVED** [review]

**Review by Javokhir12 — APPROVED** [review]

File: src/routes/ViewAllLogs/ViewAllLogs.js [hunk 1]
```diff
@@ -24,10 +22,8 @@ import {
   handleFilterChange,
 } from '@folio/stripes-acq-components';
 
-import { filterConfig } from './ViewAllLogsFilterConfig';
 import ViewAllLogsFilters from './ViewAllLogsFilters';
 import {
-  logsSearchTemplate,
   searchableIndexes,
 } from './ViewAllLogsSearchConfig';
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.js:28]: `import {  searchableIndexes } from './ViewAllLogsSearchConfig';`

File: src/routes/ViewAllLogs/ViewAllLogs.js [hunk 2]
```diff
@@ -45,8 +41,8 @@ import {
 } from '../../utils';
 import {
   FILTERS,
-  SORT_MAP,
 } from './constants';
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.js:44]: `import { FILTERS } from './constants';`

File: src/routes/ViewAllLogs/ViewAllLogs.js [hunk 3]
```diff
@@ -45,8 +41,8 @@ import {
...
+import { getQuery, getFilters, getSort } from './ViewAllLogsUtils';
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.js:45]:
```suggestion
import {
  getQuery,
  getFilters,
  getSort,
} from './ViewAllLogsUtils';
```

File: src/routes/ViewAllLogs/ViewAllLogs.js [hunk 4]
```diff
@@ -58,121 +54,49 @@ const {
...
-
-                if (filtersObj[groupIndex] === undefined) filtersObj[groupIndex] = [];
+const entityKey = 'jobLogs';
 
-                filtersObj[groupIndex] = [...filtersObj[groupIndex], ...obj[0].cql];
-              }
-            });
-          }
-        }
+export const ViewAllLogsManifest = Object.freeze({
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.js:59]: You don't need to move the manifest into a separate variable.
Refactor it next way:
1. Decorate ViewAllLogs using `@` symbol
2. Move manifest out of the class and define it like `ViewAllLogs.manifest = Object.freeze()`
3. Call manifest in tests like a static param - `ViewAllLogs.manifest`

File: src/routes/ViewAllLogs/ViewAllLogs.test.js [hunk 5]
```diff
@@ -6,11 +6,17 @@ import { noop } from 'lodash';
 
 import { renderWithIntl } from '@folio/stripes-data-transfer-components/test/jest/helpers';
 import { buildMutator } from '@folio/stripes-data-transfer-components/test/helpers';
+import { ModuleHierarchyProvider } from '@folio/stripes-core/src/components/ModuleHierarchy';
 
 import '../../../test/jest/__mock__';
-import { translationsProperties } from '../../../test/jest/helpers';
+import { translationsProperties, buildStripes } from '../../../test/jest/helpers';
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.test.js:12]:
```suggestion
import {
translationsProperties,
buildStripes,
} from '../../../test/jest/helpers';
```

File: src/routes/ViewAllLogs/ViewAllLogs.test.js [hunk 6]
```diff
@@ -6,11 +6,17 @@ import { noop } from 'lodash';
...
 
-import ViewAllLogs from './ViewAllLogs';
+import ViewAllLogs, { ViewAllLogsManifest } from './ViewAllLogs';
+import { SORT_MAP } from './constants';
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.test.js:15]:
```suggestion
import ViewAllLogs, { ViewAllLogsManifest } from './ViewAllLogs';

import { SORT_MAP } from './constants';
```

File: src/routes/ViewAllLogs/ViewAllLogs.test.js [hunk 7]
```diff
@@ -97,33 +103,70 @@ const getResources = query => ({
...
+          }}
+        />
+      </ModuleHierarchyProvider>
     </Router>
   );
 
   return renderWithIntl(component, translationsProperties);
 };
 
 // eslint-disable-next-line no-only-tests/no-only-tests
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.test.js:164]: Remove this, please

File: src/routes/ViewAllLogs/ViewAllLogs.test.js [hunk 8]
```diff
@@ -272,5 +315,157 @@ describe.skip('ViewAllLogs component', () => {
 
       expect(fileLink).toHaveAttribute('href');
     });
+
+    describe('when select logs', () => {
+      it('should render a subheading with the number of selected logs', async () => {
+        const { getAllByLabelText, getByText } = renderViewAllLogs(defaultQuery);
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.test.js:321]:
```suggestion
        const {
        getAllByLabelText,
        getByText,
        } = renderViewAllLogs(defaultQuery);
```

File: src/routes/ViewAllLogs/ViewAllLogs.test.js [hunk 9]
```diff
@@ -272,5 +315,157 @@ describe.skip('ViewAllLogs component', () => {
...
+
+        fireEvent.click(getAllByLabelText('select item')[0]);
+        fireEvent.click(getAllByLabelText('select item')[1]);
+
+        expect(getByText('2 logs selected')).toBeDefined();
+      });
+
+      it('should select all logs when click select all', async () => {
+        const { getByLabelText, getAllByLabelText } = renderViewAllLogs(defaultQuery);
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.test.js:330]: Break into separate lines if more than 1 element is destructured

File: src/routes/ViewAllLogs/ViewAllLogs.test.js [hunk 10]
```diff
@@ -272,5 +315,157 @@ describe.skip('ViewAllLogs component', () => {
...
+        fireEvent.click(getByText('Actions'));
+        fireEvent.click(getByText('Delete selected logs'));
+        fireEvent.click(getByText('Cancel'));
+
+        expect(getAllByLabelText('select item').every(checkbox => !checkbox.checked)).toBe(true);
+      });
+    });
+  });
+
+  describe('params', () => {
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.test.js:409]:
```suggestion
  describe('when filtering logs', () => {
```

File: src/routes/ViewAllLogs/ViewAllLogs.test.js [hunk 11]
```diff
@@ -272,5 +315,157 @@ describe.skip('ViewAllLogs component', () => {
...
+    describe('query', () => {
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.test.js:410]:
```suggestion
    describe('by query', () => {
```

File: src/routes/ViewAllLogs/ViewAllLogs.test.js [hunk 12]
```diff
@@ -272,5 +315,157 @@ describe.skip('ViewAllLogs component', () => {
...
+            qindex: 'hrId',
+          },
+        };
+
+        const query = ViewAllLogsManifest.records.params(null, null, queryData);
+        expect(query.hrId).toEqual(expectedQuery);
+      });
+    });
+
+    describe('sort', () => {
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.test.js:425]:
```suggestion
    describe('when sorting logs', () => {
```

File: src/routes/ViewAllLogs/ViewAllLogs.test.js [hunk 13]
```diff
@@ -272,5 +315,157 @@ describe.skip('ViewAllLogs component', () => {
...
+            sort: 'jobProfileName',
+          },
+        };
+
+        const query = ViewAllLogsManifest.records.params(null, null, queryData);
+        expect(expectedSortBy).toEqual(query.sortBy);
+      });
+    });
+
+    describe('filter', () => {
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.test.js:451]:
```suggestion
    describe('by filter params', () => {
```

File: src/routes/ViewAllLogs/ViewAllLogs.test.js [hunk 14]
```diff
@@ -272,5 +315,157 @@ describe.skip('ViewAllLogs component', () => {
...
+      it('should return hrId with the given query', () => {
+        const expectedQuery = 'testQuery*';
+        const queryData = {
+          query: {
+            query: 'testQuery',
+            qindex: 'hrId',
+          },
+        };
+
+        const query = ViewAllLogsManifest.records.params(null, null, queryData);
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.test.js:420]:
```suggestion
        const query = ViewAllLogs.manifest.records.params(null, null, queryData);
```

File: src/routes/ViewAllLogs/ViewAllLogs.js [hunk 15]
```diff
@@ -58,121 +54,49 @@ const {
...
+        ...sortValue,
+      };
+    },
+    perRequest: RESULT_COUNT_INCREMENT,
+    throwErrors: false,
+  },
+});
 @withCheckboxList
 @stripesConnect
 class ViewAllLogs extends Component {
```
Comment (mariia-aloshyna) [inline src/routes/ViewAllLogs/ViewAllLogs.js:102]:
```suggestion
@withCheckboxList
@stripesConnect
@injectIntl
export default class ViewAllLogs extends Component {
```

File: src/routes/ViewAllLogs/ViewAllLogs.js (hunk 4 above)
Comment (jakhongirabdukhamdiov) [reply src/routes/ViewAllLogs/ViewAllLogs.js:59]: I tried this version but the browser could not render it in my case, could you check if it does in your environment.

Comment (mariia-aloshyna) [reply src/routes/ViewAllLogs/ViewAllLogs.js:59]: You're right, let's just leave it like that

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**